sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_crypto.c

```diff
@@ -47,7 +47,7 @@ static cipher_t AesContext;
 
 typedef struct __attribute__((packed)) {
     uint8_t fb;
-    union {
+    union __attribute__((packed)) {
         uint32_t u8_pad;
         le_uint16_t conf_fcnt;
     };
```

Comment:
```suggestion
    uint16_t u8_pad;
    le_uint16_t conf_fcnt;
```

i am not sure why that pad is named u8_pad  maybe you are able to tell?

Reply:
Shouldn't that be

```C
    le_uint16_t conf_fcnt;
    uint16_t u8_pad;
```

Reply:
you are right 
```suggestion
    le_uint16_t conf_fcnt;
    uint16_t u16_pad;
```

Reply:
```
diff --git a/sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_crypto.c b/sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_crypto.c
index 18fc6e29b9..6bcddfde2d 100644
--- a/sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_crypto.c
+++ b/sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_crypto.c
@@ -235,11 +235,8 @@ void gnrc_lorawan_encrypt_payload(iolist_t *iolist, const le_uint32_t *dev_addr,
                                   uint32_t fcnt, uint8_t dir,
                                   const uint8_t *appskey)
 {
-    uint8_t s_block[16];
-    uint8_t a_block[16];
-
-    memset(s_block, 0, sizeof(s_block));
-    memset(a_block, 0, sizeof(a_block));
+    uint8_t s_block[16]={0};
+    uint8_t a_block[16]={0};
 
     lorawan_block0_t *block = (lorawan_block0_t *)a_block;
 
@@ -247,14 +244,11 @@ void gnrc_lorawan_encrypt_payload(iolist_t *iolist, const le_uint32_t *dev_addr,
 
     block->fb = CRYPT_B0_START;
 
-    block->u8_pad = 0;
     block->dir = dir & DIR_MASK;
 
     block->dev_addr = *dev_addr;
     block->fcnt = byteorder_htoll(fcnt);
 
-    block->u32_pad = 0;
-
     int c = 0;
 
     for (iolist_t *io = iolist; io != NULL; io = io->iol_next) {
```

Reply:
Rebased my PR. This was fixed here: https://github.com/RIOT-OS/RIOT/commit/4709bbb952c7e939cf17e4463480108f510bfa15 in PR #19634

Why was this discussed here but not there? I suspect it slipped through, as the PR had well over 1500 edits? We really should aim for small PRs when feasible :)